src/Education/XamlGenerator/RawTextProcessor.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
```

Comment:
Lots of this class was stolen from this pr -> https://github.com/SonarSource/sonarlint-visualstudio/commit/e6a0cd5089019cf2bc2e2fc0eca8ebc2deb78c42

I changed it up to instead return a list

Reply:
Don't return concrete types directly if there is an available interface i.e. return `IList` rather than `List`.

Also, use the simplest data structure possible that has the functionality required by the caller.
For example, an `IList` is writable, so returning `IList` implies it's ok for the caller to change the contents. If the result should be readonly, we can use `IReadOnlyList<>`.
Also, lists are indexable i.e. the caller can access `result[1]`. 

Here, we want the returned data to be read-only, so `IReadOnlyList<>` is an option.
The caller only really needs to iterate through the list once from the start, so `IEnumerable<>` would also do since indexing is not required.